metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/operators/datahub.py

```diff
@@ -59,5 +63,32 @@ def __init__(  # type: ignore[no-untyped-def]
         )
         self.metadata = mces
 
+    def _render_template_fields(self, field_value, context, jinja_env):
+        if isinstance(field_value, DictWrapper):
+            for key, value in field_value.items():
+                setattr(
+                    field_value,
+                    key,
+                    self._render_template_fields(value, context, jinja_env),
+                )
+        elif isinstance(field_value, list):
+            for item in field_value:
+                self._render_template_fields(item, context, jinja_env)
+        elif isinstance(field_value, str):
+            return super().render_template(field_value, context, jinja_env)
+        else:
+            return super().render_template(field_value, context, jinja_env)
+        return field_value
+
     def execute(self, context):
+        if context:
+            jinja_env = self.get_template_env()
+            for item in self.metadata:
+                if isinstance(item, MetadataChangeProposalWrapper):
+                    for key in item.__dict__.keys():
+                        value = getattr(item, key)
+                        self._render_template_fields(value, context, jinja_env)
+                if isinstance(item, MetadataChangeEvent):
+                    self._render_template_fields(item, context, jinja_env)
```

Comment:
I'm still a bit confused by this - we aren't we overriding the `render_template()` method? like why must this happen in the execute method?

Please add a comment in the code explaining the design choice